packages/scaffold-config/src/detect.ts

```diff
@@ -83,27 +83,27 @@ export function detectFramework (projectPath: string): DetectFramework {
  *     HAS TYPESCRIPT
  *   ELSE
  *     DOES NOT HAVE TYPESCRIPT
- *   ELSE IF `typescript` dependency in `package.json` OR `tsconfig.json` in `projectRoot/*`
+ *   ELSE IF `typescript` dependency in `package.json` AND `tsconfig.json` in `projectRoot/*`
  *     HAS TYPESCRIPT
  *   ELSE
  *     DOES NOT HAVE TYPESCRIPT
  *   END
  * ELSE IF HAS CYPRESS_JSON
- *   IF cypress/* contains *.ts file
+ *   IF cypress/* contains non-dts *.ts file
  *     USE TYPESCRIPT
  *   ELSE
  *     DO NOT USE TYPESCRIPT
  *   END
  * ELSE IS NEW PROJECT
- *   IF `typescript` dependency in `package.json` OR `tsconfig.json` in `projectRoot/*`
+ *   IF `typescript` dependency in `package.json` AND `tsconfig.json` in `projectRoot/*`
  *     HAS TYPESCRIPT
  *   ELSE
  *     DOES NOT HAVE TYPESCRIPT
  *   END
  * END
  */
 
-export function detectLanguage (projectRoot: string, pkgJson: PkgJson): 'js' | 'ts' {
+export function detectLanguage (projectRoot: string, pkgJson: PkgJson, isMigrating = false): 'js' | 'ts' {
```

Comment:
Consider converting this to take an object instead of a list of params. It's my prefered method instead of adding a boolean parameter. Then you get named arguments instead of a random 'true' at the end of the function and have to look it up to find out what the parameter does.(Maybe typescript helps with this?)

```js
detectLanguage({projectRoot, pkgJson: {}, isMigrating: true})
```

instead of
```js
detectLanguage(projectRoot, {} as PkgJson, true)
```